libraries/hatch-server-user-management/src/UserManagementController.ts

```diff
@@ -0,0 +1,208 @@
+import {controller, route} from '@launchtray/hatch-server';
+import {BasicRouteParams} from '@launchtray/hatch-server-middleware';
+import {inject, Logger} from '@launchtray/hatch-util';
+import 'cross-fetch/polyfill';
+import {AUTH_ACCESS_TOKEN_COOKIE_NAME} from './constants';
+import AWSCognitoClient from './AWSCognitoClient';
+import UserInfoRequest from './UserInfoRequest';
+import {TokenExpiredError} from 'jsonwebtoken';
+import {
+  AuthenticateRequest,
+  ConfirmUserRequest,
+  CreateUserRequest,
+  RefreshTokenRequest,
+  ResetPasswordRequest
+} from './UserManagementRequests';
+
+@controller()
+export default class UserManagementController {
+  
+  private authWhitelist = [
+    // default whitelist for swagger
+    '/favicon.ico',
+    '/api',
+    '/api.json',
+  ];
+  
+  constructor(
+    private readonly userService: AWSCognitoClient,
+    @inject('Logger') private readonly logger: Logger,
+    @inject('customAuthWhitelist') private readonly customAuthWhitelist: string[]) {
+    this.authWhitelist.concat(customAuthWhitelist);
```

Comment:
What happens if customAuthWhitelist is not defined / registered?

Reply:
Ideally we add an annotation that controllers can use to add route methods to the whitelist.

Reply:
Updated